composer.json

```
@@ -2,18 +2,18 @@
     "name": "mailgun/mailgun-php",
     "description": "The Mailgun SDK provides methods for all API functions.",
     "require": {
-        "php": "^7.1",
-        "psr/http-client": "^1.0",
-        "php-http/multipart-stream-builder": "^1.0",
-        "php-http/client-common": "^1.9 || ^2.0",
-        "php-http/discovery": "^1.6",
-        "webmozart/assert": "^1.6"
+        "php": "^7.3 || ^8.0",
+        "psr/http-client": "^1.0,>=1.0.1",
+        "php-http/multipart-stream-builder": "^1.0,>=1.1.2",
+        "php-http/client-common": "^2.2,>=2.2.1",
```

This would be the same
```suggestion
        "php-http/client-common": "^2.2.1",
```

---

But why do you need to specify the patch version?

---

Because the patch versions are the ones adding support for PHP 8, the previous one still has a direct link to `^7.x` instead of `>=7.x`

---

But let composer handle what exact versions to use. As long as "some" version of the dependency support PHP8 then it is fine.

---

OK, I will re-adjust it as part of the next commits, as this is still a draft. Thanks for the feedback!

---

I've implemented this change locally. I'll push it soon to the repo.